server/game/cards/03_TWI/units/TradeFederationShuttle.ts

```diff
@@ -0,0 +1,24 @@
+import AbilityHelper from '../../../AbilityHelper';
+import { NonLeaderUnitCard } from '../../../core/card/NonLeaderUnitCard';
+import { WildcardZoneName } from '../../../core/Constants';
+import { UnitCard } from '../../../core/card/CardTypes';
+
+export default class TradeFederationShuttle extends NonLeaderUnitCard {
+    protected override getImplementationId() {
+        return {
+            id: '8345985976',
+            internalName: 'trade-federation-shuttle',
+        };
+    }
+
+    public override setupCardAbilities() {
+        this.addWhenPlayedAbility({
+            title: 'create a Battle Droid token.',
+            immediateEffect: AbilityHelper.immediateEffects.conditional({ condition: (context) => context.source.controller.getUnitsInPlay(WildcardZoneName.AnyArena, (unit: UnitCard) => unit.damage > 0).length > 0,
```

Comment:
nit: can you move the condition to its own line for readability. Might actually need to be multiple lines even